shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/logging/MDCWrapper.java

```diff
@@ -0,0 +1,458 @@
+package com.redhat.service.bridge.shard.operator.logging;
+
+import java.util.Objects;
+
+import org.slf4j.Logger;
+import org.slf4j.MDC;
+import org.slf4j.Marker;
+
+import com.redhat.service.bridge.infra.models.dto.BridgeDTO;
+import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
+import com.redhat.service.bridge.shard.operator.resources.BridgeExecutor;
+import com.redhat.service.bridge.shard.operator.resources.BridgeIngress;
+
+import static com.redhat.service.bridge.shard.operator.logging.MDCCustomerIdProvider.MDC_CUSTOMER_ID_KEY;
+
+public class MDCWrapper {
+
+    private MDCWrapper() {
+        //Only static functions
+    }
+
+    public static Logger mdc(Logger logger, BridgeExecutor bridgeExecutor) {
+        return mdc(logger, bridgeExecutor.getSpec().getCustomerId());
+    }
+
+    public static Logger mdc(Logger logger, BridgeIngress bridgeIngress) {
+        return mdc(logger, bridgeIngress.getSpec().getCustomerId());
+    }
+
+    public static Logger mdc(Logger logger, BridgeDTO bridgeDTO) {
+        return mdc(logger, bridgeDTO.getCustomerId());
+    }
+
+    public static Logger mdc(Logger logger, ProcessorDTO processorDTO) {
+        return mdc(logger, processorDTO.getCustomerId());
+    }
+
+    public static Logger mdc(Logger logger, String customerId) {
+        return new MDCWrappedLogger(logger, customerId);
+    }
+
+    private static class MDCWrappedLogger implements Logger {
```

Comment:
Delegates all logging to the _real_ `Logger` but sets-up/clears MDC accordingly.